### PR TITLE
Fix FaceRecognition integration doesn't work with postgresql database.

### PR DESCRIPTION
**Pull request by matiasdelellis**
Following the discussion in #304, I tried this solution using the strict mode of mariadb/mysql.

Split the Clusters and People queries to minimize the considerations of both functions. Otherwise I had to add many nested ifs.

Also change to only show 15 clusters (as a small optimization) since there is no point in showing everything, and it is better to show these little by little.

### CODE REVIEW

File: lib/Db/TimelineQueryPeopleFaceRecognition.php [hunk 1]
```diff
@@ -214,6 +162,113 @@ public function getFaceRecognitionPreview(TimelineRoot &$root, $currentModel, $p
...
+            $query->expr()->eq('fri.file', 'm.fileid'),
+            $query->expr()->eq('fri.model', $query->createNamedParameter($currentModel)),
+        ));
+
+        // WHERE these photos are in the user's requested folder recursively
+        $query = $this->joinFilecache($query, $root, true, false);
+
+        // GROUP by ID of face cluster
+        $query->groupBy('frp.id');
+        $query->addGroupBy('frp.user');
```
Comment (matiasdelellis) [inline lib/Db/TimelineQueryPeopleFaceRecognition.php:192]: This particular line doesn't make sense to me, but it seems that in strict mode, mariadb doesn't like that all rows ar equal and forces grouping for it as well. 🤔

File: lib/Db/TimelineQueryPeopleFaceRecognition.php [hunk 2]
```diff
@@ -214,6 +162,113 @@ public function getFaceRecognitionPreview(TimelineRoot &$root, $currentModel, $p
...
+            $query->expr()->eq('fri.file', 'm.fileid'),
+            $query->expr()->eq('fri.model', $query->createNamedParameter($currentModel)),
+        ));
+
+        // WHERE these photos are in the user's requested folder recursively
+        $query = $this->joinFilecache($query, $root, true, false);
+
+        // GROUP by name of face clusters
+        $query->where($query->expr()->isNotNull('frp.name'));
+        $query->groupBy('frp.user');
```
Comment (matiasdelellis) [inline lib/Db/TimelineQueryPeopleFaceRecognition.php:252]: Same here..

File: lib/Db/TimelineQueryPeopleFaceRecognition.php (hunk 1 above)
Comment (matiasdelellis) [reply lib/Db/TimelineQueryPeopleFaceRecognition.php:192]: The integration with recognize in strict mode also fails with:

```
[index] Error: OC\DB\Exceptions\DbalException: An exception occurred while executing a query: SQLSTATE[42000]: Syntax error or access violation: 1055 'nextcloud.rfc.user_id' isn't in GROUP BY at <<closure>>

0. /var/www/html/nextcloud/lib/private/DB/ConnectionAdapter.php line 69
   OC\DB\Exceptions\DbalException::wrap(Doctrine\DBAL\Ex ... "})
1. /var/www/html/nextcloud/apps/memories/lib/Db/TimelineQueryDays.php line 328
   OC\DB\ConnectionAdapter->executeQuery("WITH RECURSIVE  ... C", [[44623],false], [101,5])
2. /var/www/html/nextcloud/apps/memories/lib/Db/TimelineQueryPeopleRecognize.php line 75
   OCA\Memories\Db\TimelineQuery->executeQueryWithCTEs(OC\DB\QueryBuilder\QueryBuilder {})
3. /var/www/html/nextcloud/apps/memories/lib/Controller/PeopleController.php line 59
   OCA\Memories\Db\TimelineQuery->getPeopleRecognize(OCA\Memories\Db\TimelineRoot {})
4. /var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 225
   OCA\Memories\Controller\PeopleController->recognizePeople()
5. /var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 133
   OC\AppFramework\Http\Dispatcher->executeController(OCA\Memories\Con ... {}, "recognizePeople")
6. /var/www/html/nextcloud/lib/private/AppFramework/App.php line 172
   OC\AppFramework\Http\Dispatcher->dispatch(OCA\Memories\Con ... {}, "recognizePeople")
7. /var/www/html/nextcloud/lib/private/Route/Router.php line 298
   OC\AppFramework\App::main("OCA\\Memories\\ ... r", "recognizePeople", OC\AppFramework\ ... {}, ["memories.People.recognizePeople"])
8. /var/www/html/nextcloud/lib/base.php line 1047
   OC\Route\Router->match("/apps/memories/api/recognize/people")
9. /var/www/html/nextcloud/index.php line 36
   OC::handleRequest()

GET /apps/memories/api/recognize/people
from 190.48.188.12 by user at 2023-02-03T15:21:38+00:00
```

So, if it was never reported before, it is probably an option that I activate in mariadb, and this line is not necessary.. 🤔

**Comment by pulsejet** [comment]
LGTM

**Pull request merged**